nexus/reconfigurator/planning/src/system.rs

```diff
@@ -828,3 +823,27 @@ impl Sled {
         &self.inventory_sled_agent
     }
 }
+
+#[derive(Clone, Copy, Debug)]
+struct SubnetIterator {
```

Comment:
If we clone this, do we want the clone to get the iterator at its current position (which I _think_ is what we get here?) or the iterator in its original position?

Reply:
Current position, since this is a wrapper around a stateful iterator. Current position is what we want, from the perspective of reconfigurator simulation.